ray-operator/config/samples/ray-data-image-resize/ray_data_image_resize.py

```diff
@@ -2,30 +2,64 @@
 
 import numpy as np
 import ray
+import io
+import os
+
+import torch
 from torchvision import transforms
 
-DATA_URI = "gs://anonymous@ray-images/images"
+from google.cloud import storage
+
+bucket_name = os.environ["BUCKET_NAME"]
+prefix = os.environ["BUCKET_PREFIX"]
+allowed_extensions = ('.png', '.jpg', '.jpeg', '.tif', '.tiff', '.bmp', '.gif')
+
+def list_blobs(bucket_name, prefix):
+    client = storage.Client().create_anonymous_client()
+    bucket = client.bucket(bucket_name)
+    blobs = bucket.list_blobs(prefix=prefix)
+
+    blob_files = []
+    for blob in blobs:
+        if blob.name.lower().endswith(allowed_extensions):
+            blob_files.append(blob.name)
+    return blob_files
+
+def download_blob(blob_name):
+    client = storage.Client().create_anonymous_client()
+    bucket = client.bucket(bucket_name)
+    blob = bucket.blob(blob_name['item'])
+    data = blob.download_as_bytes()
+
+    from PIL import Image
+    Image.MAX_IMAGE_PIXELS = None
+    image = np.array(Image.open(io.BytesIO(data)))
+
+    return {"image": image}
+
+def transform_frame(row: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:
+    transform = transforms.Compose(
+        [transforms.ToTensor(), transforms.Resize((256, 256)), transforms.ConvertImageDtype(torch.float)]
+    )
+    row["image"] = transform(row["image"])
+    return row
 
 def main():
     """
     This is a CPU-only job that reads images from
     Google Cloud Storage and resizes them.
     """
     ray.init()
-    dataset = ray.data.read_images(DATA_URI, include_paths=True)
```

Comment:
I ran into issues with read_images because I was hitting the pixel size limit with some larger datasets:
```
  File "/home/ray/anaconda3/lib/python3.8/site-packages/ray/data/datasource/image_datasource.py", line 81, in _read_stream
    image = Image.open(io.BytesIO(data))
  File "/home/ray/anaconda3/lib/python3.8/site-packages/PIL/Image.py", line 3133, in open
    im = _open_core(fp, filename, prefix, formats)
  File "/home/ray/anaconda3/lib/python3.8/site-packages/PIL/Image.py", line 3120, in _open_core
    _decompression_bomb_check(im.size)
  File "/home/ray/anaconda3/lib/python3.8/site-packages/PIL/Image.py", line 3029, in _decompression_bomb_check
    raise DecompressionBombError(
PIL.Image.DecompressionBombError: Image size (222926661 pixels) exceeds limit of 178956970 pixels, could be decompression bomb DOS attack.
```

Worked around this by using `ray.data.from_items` and then doing a `map` that downloads the images with `Image.MAX_IMAGE_PIXELS = None` to avoid the pixel size limit. 

I also opened https://github.com/ray-project/ray/pull/45415 so we can update this to use `ray.data.read_images` again

Reply:
Yes, ideally we should use `ray.data.read_images` after the issue is fixed.